lib/ukdebug/README.md

```diff
@@ -0,0 +1,154 @@
+# `ukdebug`: Unikraft's debugging features
+
+This library contains features to debug Unikraft.
+
+## GDB stub
+
+The GDB stub allows remotely debugging Unikraft with GDB using the [GDB Remote Serial Protocol (RSP)](https://sourceware.org/gdb/current/onlinedocs/gdb.html/Remote-Protocol.html).
+The debugging process involves two machines.
+A development machine (the GDB host) that runs GDB and implements the client side of the RSP.
+The other machine runs Unikraft (the target), which implements the GDB stub and serves requests issued by the host.
+The two sides communicate over a serial connection.
+With this setup, the machine running Unikraft can be a local VM or a physically remote machine.
+
+```plain
+   Development machine                     Unikraft
+  ┌─────────────┐                         ┌─────────────┐
+  │             │    Serial connection    │             │
+  │  GDB Host ◄─┼─────────────────────────┼─► GDB Stub  │
+  │             │                         │             │
+  └─────────────┘                         └─────────────┘
```

Comment:
Use ASCII characters for this pls

Reply:
@StefanJum can you please elaborate for this requirement?

Reply:
(As discussed in the maintainers meeting)
1. We can be sure ASCII characters are rendered properly everywhere, no matter the layout/language etc.
2. The README files will eventually be part of the documentation website, were we already use the ASCII-only convention, so it makes sense for consistency reasons.

Reply:
Fair points. I switched the graphic to ASCII :+1: